src/lib/hooks/useActiveNetwork.ts

```diff
@@ -0,0 +1,23 @@
+import { useLocation } from 'react-router-dom';
+import { WalletNetwork } from '@/lib/store/wallet';
+import { useProfileContext } from '@/lib/context/Profile';
+
+const useActiveNetwork = () => {
+    const { profile } = useProfileContext();
+
+    const { state } = useLocation();
+
+    let selectedNetwork = state?.isTestnet ? WalletNetwork.DEVNET : WalletNetwork.MAINNET;
+
+    if (profile.wallets().count() === 0) {
+        selectedNetwork = WalletNetwork.MAINNET;
+    }
+
+    console.log(selectedNetwork);
```

Comment:
For reviewers

Reply:
better to remove this and leave a comment instead where they could add it; saves accidental merges with `console.log` statements + it can be merged right after reviews instead of waiting for an additional commit